File: docs/snippets/go/docker/usage.go
```go
package main

import (
	"bytes"
	"encoding/json"
	"fmt"
	"io"
	"mime/multipart"
	"net/http"
	"os"
	"os/exec"
	"path/filepath"
	"time"
)

type DockerKreuzbergClient struct {
	containerName string
	containerImage string
	apiPort       int
}

func NewDockerKreuzbergClient(containerName, image string, port int) *DockerKreuzbergClient {
	return &DockerKreuzbergClient{
		containerName: containerName,
		containerImage: image,
		apiPort:       port,
	}
}

func (c *DockerKreuzbergClient) StartContainer() error {
	fmt.Println("Starting Kreuzberg Docker container...")
	cmd := exec.Command("docker", "run", "-d",
		"--name", c.containerName,
		"-p", fmt.Sprintf("%d:8000", c.apiPort),
		c.containerImage)

	if err := cmd.Run(); err != nil {
		return fmt.Errorf("failed to start container: %w", err)
	}

	fmt.Printf("Container started on http://localhost:%d\n", c.apiPort)
	return nil
}

func (c *DockerKreuzbergClient) ExtractFile(filePath string) (string, error) {
	fileBytes, err := os.ReadFile(filePath)
	if err != nil {
		return "", err
	}

	var buf bytes.Buffer
	writer := multipart.NewWriter(&buf)

	part, err := writer.CreateFormFile("file", filepath.Base(filePath))
	if err != nil {
		return "", err
	}

	if _, err := io.Copy(part, bytes.NewReader(fileBytes)); err != nil {
		return "", err
	}

	if err := writer.Close(); err != nil {
		return "", err
	}

	resp, err := http.Post(
		fmt.Sprintf("http://localhost:%d/api/extract", c.apiPort),
		writer.FormDataContentType(),
		&buf,
	)
	if err != nil {
		return "", err
	}
	defer resp.Body.Close()

	var result map[string]string
	if err := json.NewDecoder(resp.Body).Decode(&result); err != nil {
		return "", err
	}

	return result["content"], nil
}

func (c *DockerKreuzbergClient) StopContainer() error {
	fmt.Println("Stopping Kreuzberg Docker container...")
	if err := exec.Command("docker", "stop", c.containerName).Run(); err != nil {
		return err
	}
	if err := exec.Command("docker", "rm", c.containerName).Run(); err != nil {
		return err
	}
	fmt.Println("Container stopped and removed")
	return nil
}

func main() {
	client := NewDockerKreuzbergClient("kreuzberg-api", "kreuzberg:latest", 8000)

	if err := client.StartContainer(); err != nil {
		panic(err)
	}

	time.Sleep(2 * time.Second)

	content, err := client.ExtractFile("document.pdf")
	if err != nil {
		panic(err)
	}

	fmt.Printf("Extracted content:\n%s\n", content)

	if err := client.StopContainer(); err != nil {
		panic(err)
	}
}
```

File: docs/snippets/rust/docker/usage.rs
```rust
use std::process::Command;
use anyhow::Result;
use reqwest::multipart;
use std::fs;

struct DockerKreuzbergClient {
    container_name: String,
    api_port: u16,
}

impl DockerKreuzbergClient {
    fn new(container_name: &str, api_port: u16) -> Self {
        Self {
            container_name: container_name.to_string(),
            api_port,
        }
    }

    fn start_container(&self, image: &str) -> Result<()> {
        println!("Starting Kreuzberg Docker container...");
        let status = Command::new("docker")
            .args(&[
                "run",
                "-d",
                "--name",
                &self.container_name,
                "-p",
                &format!("{}:8000", self.api_port),
                image,
            ])
            .status()?;

        if !status.success() {
            anyhow::bail!("Failed to start Docker container");
        }

        println!("Container started on http://localhost:{}", self.api_port);
        Ok(())
    }

    async fn extract_file(&self, file_path: &str) -> Result<String> {
        let file_bytes = fs::read(file_path)?;
        let file_part = multipart::Part::bytes(file_bytes)
            .file_name(file_path.to_string());
        let form = multipart::Form::new()
            .part("file", file_part);

        let client = reqwest::Client::new();
        let response = client
            .post(&format!("http://localhost:{}/api/extract", self.api_port))
            .multipart(form)
            .send()
            .await?;

        let json: serde_json::Value = response.json().await?;
        Ok(json["content"].as_str().unwrap_or("").to_string())
    }

    fn stop_container(&self) -> Result<()> {
        println!("Stopping Kreuzberg Docker container...");
        Command::new("docker")
            .args(&["stop", &self.container_name])
            .status()?;
        Command::new("docker")
            .args(&["rm", &self.container_name])
            .status()?;
        println!("Container stopped and removed");
        Ok(())
    }
}

#[tokio::main]
async fn main() -> Result<()> {
    let docker_client = DockerKreuzbergClient::new("kreuzberg-api", 8000);

    docker_client.start_container("kreuzberg:latest")?;
    tokio::time::sleep(tokio::time::Duration::from_secs(2)).await;

    let content = docker_client.extract_file("document.pdf").await?;
    println!("Extracted content:\n{}", content);

    docker_client.stop_container()?;
    Ok(())
}
```
